dropwizard-core/src/test/java/io/dropwizard/server/AbstractServerFactoryTest.java

```diff
@@ -0,0 +1,103 @@
+package io.dropwizard.server;
+
+import io.dropwizard.Configuration;
+import io.dropwizard.jersey.DropwizardResourceConfig;
+import io.dropwizard.jersey.setup.JerseyContainerHolder;
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import io.dropwizard.setup.Environment;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.thread.ThreadPool;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests that the {@link JerseyEnvironment#getUrlPattern()} is set by the following priority order:
+ * <ol>
+ *     <li>YAML defined value</li>
+ *     <li>{@link io.dropwizard.Application#run(Configuration, Environment)} defined value</li>
+ *     <li>Default value defined by {@link DropwizardResourceConfig#urlPattern}</li>
+ * </ol>
+ */
+public class AbstractServerFactoryTest {
+
+    private final JerseyContainerHolder holder = mock(JerseyContainerHolder.class);
+    private final DropwizardResourceConfig config = new DropwizardResourceConfig();
+    private final JerseyEnvironment jerseyEnvironment = new JerseyEnvironment(holder, config);
+    private final Environment environment = mock(Environment.class, RETURNS_DEEP_STUBS);
+    private final TestServerFactory serverFactory = new TestServerFactory();
+
+    private static final String DEFAULT_PATTERN = "/*";
+    private static final String RUN_SET_PATTERN = "/set/from/run/*";
+    private static final String YAML_SET_PATTERN = "/set/from/yaml/*";
+
+    @Before
+    public void before() {
+        when(this.environment.jersey()).thenReturn(this.jerseyEnvironment);
```

Comment:
I think you could avoid to use `this` to reference the fields in the unit test, because there are no name conflicts.